utils/S3UtilsMongoClient.js

```diff
@@ -261,6 +357,23 @@ class S3UtilsMongoClient extends MongoClientInterface {
             const retResult = this._handleResults(collRes, isVer);
             retResult.stalled = stalledCount;
 
+            if (inflightsPreScan > 0 && retResult && retResult.dataMetrics) {
+                if (Object.keys(retResult.dataMetrics.bucket).length > 0) {
+                    Object.keys(retResult.dataMetrics.bucket).forEach(key => {
+                        retResult.dataMetrics.bucket[key] = {
+                            ...retResult.dataMetrics.bucket[key],
+                            usedCapacity: {
+                                ...retResult.dataMetrics.bucket[key].usedCapacity,
+                                _inflightsPreScan: inflightsPreScan,
+                            },
+                        };
+                        Object.keys(retResult.dataMetrics.bucket).forEach(() => {
+                            retResult.dataMetrics.bucket[key].accountOwnerID = accountBucket;
+                        });
+                    });
+                }
+            }
+
             return callback(null, retResult);
         } catch (err) {
```

Comment:
weird to iterate other each bucket inside the same iteration, and call repeatedly the same function...

did you mean:

```suggestion
                            accountOwnerID: accountBucket;
                        });
```